cfg.go

```diff
@@ -61,9 +64,77 @@ func (c *configWrapper) mergeEnv(envName string) error {
 	return c.v.MergeInConfig()
 }
 
+func (c *configWrapper) loadEnvConfigFile() {
+	tempV := viper.New()
+
+	tempV.SetConfigType(c.params.FileType)
+	tempV.SetConfigName(envVarConfigFile)
+	tempV.AddConfigPath(c.params.Path)
+	if err := tempV.ReadInConfig(); err != nil {
+		return
+	}
+	keys := tempV.AllKeys()
+	for _, key := range keys {
+		envVarName := tempV.GetString(key)
+		if value := os.Getenv(envVarName); value != "" {
+			if isBool(value) {
+				bValue, _ := strconv.ParseBool(value)
+				c.v.Set(key, bValue)
+			} else if isInt(value) {
+				iValue, _ := strconv.Atoi(value)
+				c.v.Set(key, iValue)
+			} else if isFloat(value) {
+				fValue, _ := strconv.ParseFloat(value, 64)
+				c.v.Set(key, fValue)
+			} else {
+				c.v.Set(key, value)
+			}
+		}
+	}
+	return
+}
+
 // Get returns a map[string]interface{} of a given key
-func (c *configWrapper) Get(key string) map[string]interface{} {
-	return c.v.GetStringMap(key)
+func (c *configWrapper) Get(key string) interface{} {
```

Comment:
There is some edge cases where I think this implementation does not work.  The following testcase fails: 

```"TestNonExistentConfig": {
			key:               "services.foo.string.foo",
			expectedConfigMap: nil,
		},
```

But don't reinvent the wheel, this logic is already implemented in the [Viper](https://godoc.org/github.com/spf13/viper) library. Simply do `return c.v.Get(key)`

Reply:
c.v.Get does not returns the full map after overriding it with `Set()` 😢